### PR TITLE
Added file rename util.

### PR DESCRIPTION
**Pull request by mickel1138**
Added new rename file util, and while loop to ensure file doesn't exist, even after being renamed.

### CODE REVIEW

File: src/Controller/Api/FileManager.php [hunk 1]
```diff
@@ -6,9 +6,15 @@
 use App\Entity\File;
 use App\Entity\Fileset;
 use App\Util\Datastore;
+use App\Util\RenameDuplicate;
+use App\Util\FileUploader;
```
Comment (mickel1138) [inline src/Controller/Api/FileManager.php:10]: Flip

File: src/Controller/Api/FileManager.php [hunk 2]
```diff
@@ -19,6 +25,60 @@
...
+        $this->denyAccessUnlessGranted('IS_AUTHENTICATED_FULLY');
+        $fileMetadata = $fileUploader->combineChunks($request);
+
+        $fileName = $fileMetadata['name'];
+        $filePath = $fileMetadata['path'];
+        $fileSize = $fileMetadata['size'];
+
+        $fileset = $datasetSubmission->getFileset();
+
+        while ($fileset->doesFileExist($fileName)) {
```
Comment (praneethpr) [inline src/Controller/Api/FileManager.php:58]: This should be moved inside the else part of if Fileset exists or not

File: src/Util/RenameDuplicate.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        return $fileName;
+    }
+
+    /**
+     * Renames the file and adds a sequence to it.
+     *
+     * @param array $matches The match components of the regex.
+
+     * @return string The renamed filename string.
+     *
```
Comment (praneethpr) [inline src/Util/RenameDuplicate.php:44]: Extra line

File: src/Util/RenameDuplicate.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+     */
+    private function addSequence(array $matches) : string
+    {
+        // If there are 3 matches, then the filename does not contain a sequence yet.
+        if (count($matches) === 3) {
+            return $matches[1].'(1)'.$matches[2];
+        } else {
+            if ($matches[2] >= 999) {
```
Comment (praneethpr) [inline src/Util/RenameDuplicate.php:52]: Type comparison

File: src/Controller/Api/FileManager.php [hunk 5]
```diff
@@ -6,9 +6,15 @@
...
 use App\Util\Datastore;
+use App\Util\RenameDuplicate;
+use App\Util\FileUploader;
+
 use Doctrine\ORM\EntityManagerInterface;
+
 use FOS\RestBundle\Controller\AbstractFOSRestController;
+
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
+
```
Comment (praneethpr) [inline src/Controller/Api/FileManager.php:17]: All blank lines can be removed(PSR-12)

**Review by praneethpr — COMMENTED** [review]
Tested, and works fine

**Review by praneethpr — APPROVED** [review]

File: src/Controller/Api/FileManager.php [hunk 6]
```diff
@@ -19,6 +21,64 @@
...
 {
+    /**
+     * Adds a file to a dataset submission.
+     *
+     * @param DatasetSubmission      $datasetSubmission The id of the dataset submission.
+     * @param Request                $request           The request body sent with file metadata.
+     * @param EntityManagerInterface $entityManager     Entity manager interface to doctrine operations.
+     * @param FileUploader           $fileUploader      File upload handler service.
+     * @param RenameDuplicate        $renameDuplicate   The duplicate renaming utility.
+     *
```
Comment (mickel1138) [inline src/Controller/Api/FileManager.php:32]: Add `@throw`

**Review by praneethpr — APPROVED** [review]

**Pull request merged**